server/sonar-web/src/main/js/apps/projects/components/AllProjects.js

```diff
@@ -28,112 +30,116 @@ import { parseUrlQuery } from '../store/utils';
-  };
+type Props = {
+  isFavorite: boolean,
+  location: { pathname: string, query: { [string]: string } },
+  fetchProjects: (query: string, isFavorite: boolean, organization?: {}) => Promise<*>,
+  organization?: { key: string },
+  router: { push: ({ pathname: string, query?: {} }) => void }
+};
+
+type State = {
+  query: { [string]: string },
+  optionBarOpen: boolean
+};
 
-  state = {
-    query: {}
+export default class AllProjects extends React.PureComponent {
+  props: Props;
+  state: State = {
+    query: {},
+    optionBarOpen: false
   };
 
   componentDidMount() {
     this.handleQueryChange();
-    document.getElementById('footer').classList.add('search-navigator-footer');
+    const footer = document.getElementById('footer');
+    footer && footer.classList.add('search-navigator-footer');
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps: Props) {
     if (prevProps.location.query !== this.props.location.query) {
       this.handleQueryChange();
     }
   }
 
   componentWillUnmount() {
-    document.getElementById('footer').classList.remove('search-navigator-footer');
+    const footer = document.getElementById('footer');
+    footer && footer.classList.remove('search-navigator-footer');
   }
 
-  handleQueryChange() {
-    const query = parseUrlQuery(this.props.location.query);
-    this.setState({ query });
-    this.props.fetchProjects(query, this.props.isFavorite, this.props.organization);
-  }
+  handleOptionBarToggle = (open: boolean) => this.setState({ optionBarOpen: open });
 
-  handleViewChange = view => {
-    const query = {
-      ...this.props.location.query,
-      view: view === 'list' ? undefined : view
-    };
-    if (query.view !== 'visualizations') {
-      Object.assign(query, { visualization: undefined });
-    }
-    this.props.router.push({
-      pathname: this.props.location.pathname,
-      query
-    });
-  };
-
-  handleVisualizationChange = visualization => {
+  handlePerspectiveChange = ({ view, visualization }: { view: string, visualization?: string }) => {
     this.props.router.push({
       pathname: this.props.location.pathname,
       query: {
         ...this.props.location.query,
-        view: 'visualizations',
+        view: view === 'overall' ? undefined : view,
         visualization
       }
     });
   };
 
+  handleQueryChange() {
+    const query = parseUrlQuery(this.props.location.query);
+    this.setState({ query });
+    this.props.fetchProjects(query, this.props.isFavorite, this.props.organization);
+  }
+
   render() {
-    const { query } = this.state;
+    const { query, optionBarOpen } = this.state;
     const isFiltered = Object.keys(query).some(key => query[key] != null);
 
-    const view = query.view || 'list';
+    const view = query.view || 'overall';
     const visualization = query.visualization || 'risk';
 
-    const top = this.props.organization ? 95 : 30;
+    const top = (this.props.organization ? 95 : 30) + (optionBarOpen ? 45 : 0);
```

Comment:
Ah yes, good catch! Or maybe the top bar should be fixed ? And not hidden when scrolling ?